public/app/features/geo/utils/frameVectorSource.ts

```diff
@@ -57,4 +57,59 @@ export class FrameVectorSource<T extends Geometry = Geometry> extends VectorSour
     // only call this at the end
     this.changed();
   }
+
+  updateEdge(frames: DataFrame[]) {
```

Comment:
hymm -- something feels wrong here.  this specific network processing most likely belongs with the network layer

Reply:
i think it could actually just be a function defined there :)